src/cart/services/cart.service.ts

```diff
@@ -1,31 +1,57 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import fetch from 'node-fetch';
 import { Injectable } from '@nestjs/common';
-
+import getDBClient from 'src/database/utils/getDBClient';
 import { v4 } from 'uuid';
 
-import { Cart } from '../models';
+import { Cart, CartItem } from '../models';
+import { PoolClient } from 'pg';
 
 @Injectable()
 export class CartService {
-  private userCarts: Record<string, Cart> = {};
-
-  findByUserId(userId: string): Cart {
-    return this.userCarts[ userId ];
+  async findByUserId(userId: string): Promise<Cart> {
+    const dbClient = await getDBClient();
+    const cart = await dbClient.query(
+      'SELECT c.id, ci.product_id, ci.count FROM carts as c inner join cart_items as ci on c.id = ci.cart_id where user_id = $1;',
+      [userId]
+    );
+    dbClient.end();
+
+    const productsDataResponse = await fetch('https://6b1jcnr1b2.execute-api.eu-west-1.amazonaws.com/products');
```

Comment:
Not the best way to get data, better to create query with joining tables and get data from DB directly, or hide URL to env variables at least.